### PR TITLE
Fix for TriggerMode.HOTKEY (3) showing up in modes in the save file t…

### PR DESCRIPTION
**Pull request by sebastiansam55**
…wice.

Previously saving would add 3, and only 3, twice to the modes portion of the save file.

And fix for the preferences window not saving changes

### CODE REVIEW

File: lib/autokey/model/abstract_hotkey.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def set_hotkey(self, modifiers, key):
         modifiers.sort()
         self.modifiers = modifiers
         self.hotKey = key
-        if key is not None:
+        if key is not None and TriggerMode.HOTKEY not in self.modes:
```
Comment (BlueDrink9) [inline lib/autokey/model/abstract_hotkey.py:48]: This change specifically breaks some tests relating to creating phrases with duplicate hotkeys (which is forbidden). Normally creating duplicates raises an error, but with this change it does not. Could you please review this again @sebastiansam55

**Review by BlueDrink9 — COMMENTED** [review]
@sebastiansam55 I forgot to be explicit when you started making PRs, but you should be running the tests please

**Comment by sebastiansam55** [comment]
Noted will do so going forward!

**Comment by BlueDrink9** [comment]
Thank you. Once I've fixed this bug, I'll push a note to my branch on how to add automatically running the tests as a pre-push git hook. That's how I caught these failures.

**Comment by BlueDrink9** [comment]
What was the actual issue here? I think I have solved it a different way, because the way it was solved here was causing problems elsewhere and I have reverted it on my branch. I'd like to test/confirm my fix, but I'm not totally sure what the issue description was @sebastiansam55

**Comment by sebastiansam55** [comment]
From what I remember;
My `.json` files started showing up with two 3's in the `modes` list. This fix made it stop for me.

I probably should have waited longer before starting to open commits; this is my first return to any form of programming for about 5 years so I probably misunderstood some stuff.

**Comment by BlueDrink9** [comment]
I think the only issue was how little personpower for checking PRs we have for this repo specifically. Commits in general are welcome.

**Pull request merged**